src/main/java/org/spongepowered/api/block/data/Beacon.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.block.data;
+
+import org.spongepowered.api.potion.PotionEffectType;
+
+/**
+ * Represents a Beacon.
+ */
+public interface Beacon extends Lockable {
+
+    /**
+     * Gets the primary effect provided by this beacon.
+     * 
+     * @return The primary effect
+     */
+    PotionEffectType getPrimaryEffect();
+
+    /**
+     * Sets the primary effect for this beacon.
+     * 
+     * @param effect The new primary effect
+     */
+    void setPrimaryEffect(PotionEffectType effect);
```

Comment:
In vanilla minecraft the effects are quite limited to specific types, is this an UI limitation only or is this a forced/fixed limitation? If it is forced/fixed, you should add either an exception, a return value or at least some javadocs that it might not set the `PotionEffectType`. Maybe a method `getAllowedEffects()` would be good in that case as well.